src/Raven.Server/Documents/Indexes/Workers/MapItems.cs

```diff
@@ -138,9 +138,9 @@ protected MapItems(Index index, IndexStorage indexStorage, MapReduceIndexingCont
                                     try
                                     {
                                         current.KnownToBeNew = _indexStorage.LowerThanLastDatabaseEtagOnIndexCreation(current.Etag);
-                                        
-                                        if ((itemEnumerator.Current.DocumentFlags.HasFlag(DocumentFlags.Archived) == false && _index.Definition.SourceItemKind != IndexSourceItemKind.ArchivedOnly) ||
-                                            (itemEnumerator.Current.DocumentFlags.HasFlag(DocumentFlags.Archived) == true && _index.Definition.SourceItemKind != IndexSourceItemKind.Default))
+                                        if (_index.SourceType != IndexSourceType.Documents || 
```

Comment:
let me know if this change is ok
regarding we won't care about ItemKind (set to null earlier)

Reply:
simplified the logic with arek, resolving this conversation